src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import styles from './TextArea.st.css';
+import { TEXT_AREA_DATA_HOOK, TEXT_AREA_ERROR_DATA_HOOK } from './dataHooks';
+import { Tooltip } from '../Tooltip';
+import { ReactComponent as ErrorIcon } from '../../assets/icons/Error.svg';
+import {
+  TEXT_AREA_DISABLED,
+  TEXT_AREA_SUCCESS,
+  TEXT_AREA_ERROR,
+  TEXT_AREA_THEME,
+} from './dataKeys';
+
+export enum TextAreaTheme {
+  Box = 'box',
+  Line = 'line',
+}
+
+export interface TextAreaProps {
+  value: string;
+  error?: boolean;
+  success?: boolean;
+  dataHook?: string;
+  ariaLabel: string;
+  disabled?: boolean;
+  placeholder?: string;
+  /** Possible values: 'line', 'box' */
+  theme?: TextAreaTheme;
+  errorDescription?: string;
+  onChange(event: React.ChangeEvent<HTMLTextAreaElement>): void;
+}
+
+interface DefaultProps {
+  error: boolean;
+  success: boolean;
+  disabled: boolean;
+  dataHook: string;
+  placeholder: string;
+  theme: TextAreaTheme;
+  errorDescription: string;
+}
+
+interface State {}
+
+/** TextArea component */
+export class TextArea extends React.Component<TextAreaProps, State> {
+  static displayName = 'TextArea';
+  static defaultProps: DefaultProps = {
+    dataHook: '',
+    error: false,
+    success: false,
+    disabled: false,
+    placeholder: '',
+    errorDescription: '',
+    theme: TextAreaTheme.Box,
+  };
+
+  _getDataAttributes() {
+    const { disabled, success, theme, error } = this.props;
+
+    return {
+      [TEXT_AREA_DISABLED]: disabled,
+      [TEXT_AREA_THEME]: theme,
+      [TEXT_AREA_ERROR]: error,
+      [TEXT_AREA_SUCCESS]: success,
+    };
+  }
+
+  render() {
+    const {
+      value,
+      theme,
+      error,
+      success,
+      dataHook,
+      disabled,
+      onChange,
+      placeholder,
+      errorDescription,
+      ...rest
+    } = this.props;
+    const dataObject = this._getDataAttributes();
+    const showErrorIcon = error && errorDescription;
+    return (
+      <div
+        data-hook={dataHook}
+        {...dataObject}
+        {...styles(
+          'root',
+          {
+            error,
+            theme,
+            success,
+            disabled,
+          },
+          rest,
+        )}
+      >
+        <textarea
+          value={value}
+          onChange={e => {
+            if (!disabled) {
```

Comment:
This is actually not needed.
Instead we should have a `disabled={disabled}` attribute.